ddp-workspace/projects/ddp-sdk/src/lib/components/languageSelector.component.ts

```diff
@@ -73,43 +85,133 @@ export class LanguageSelectorComponent implements OnInit, OnDestroy {
   }
 
   public changeLanguage(lang: StudyLanguage): void {
+    if (!isNullOrUndefined(this.currentLanguage) && this.currentLanguage.languageCode === lang.languageCode) {
+      return;
+    }
+
     if (this.language.canUseLanguage(lang.languageCode)) {
       this.currentLanguage = lang;
-      this.language.changeLanguage(lang.languageCode);
+      if (this.language.getCurrentLanguage() !== lang.languageCode) {
+        this.language.changeLanguage(lang.languageCode);
+        if (this.session.isAuthenticatedSession()) {
+          this.updateProfileLanguage();
+        }
+      }
     }
     else {
       console.error('Error: The specified language: ' + JSON.stringify(lang) + ' is not configured for the study.');
     }
   }
 
+  //Update the language in the profile to the current language
+  private updateProfileLanguage(): void {
+    // Create observable that gets the user's profile and creates a copy with updated preferred language
+    let getProfileObservable: Observable<UserProfile> = this.profileServiceAgent.profile
+      .pipe(map(x => {
+        return {...x.profile, preferredLanguage: this.currentLanguage.languageCode};
+      })).pipe(take(1));
+
+    // When we get the new profile value, save it
+    let updateProfileObservable: Observable<any> = getProfileObservable
+      .pipe(concatMap(profile => this.profileServiceAgent.saveProfile(false, profile)));
```

Comment:
Since this `saveProfile()` call is really just doing a `PATCH` call, you can make API request with just the new language code. This way, you don't need to do a fetch beforehand.